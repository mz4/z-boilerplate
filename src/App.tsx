import React from "react";
import { useState } from "react";
import Title from "./components/Title";
import playersL from "./components/players.json";
import RenderForm from "./components/RenderForm";
import Boxes from "./components/Boxes";

interface IState {
  players: {
    id: number;
    name: string;
    role: string;
    age: number;
    active: boolean;
  }[];
}

const App: React.FC<{}> = () => {
  const [players, playersList] = useState<IState["players"]>(playersL);
  const [modal, showModal] = useState(false);

  const deletePlayer = (id: number) => {
    playersList(players.filter((p) => p.id !== id));
  };

  const closeModal = () => {
    showModal(false);
  };

  const setPlayer = (players: any) => {
    playersList(players);
  };

  const addNewPlayer = () => {
    showModal(!modal);
  };

  return (
    <>
      <Boxes />
      <Title players={players} deletePlayer={deletePlayer} />
      <button onClick={addNewPlayer} data-testid="addPlayer">
        Add Player
      </button>
      {modal ? (
        <RenderForm
          players={players}
          setPlayer={setPlayer}
          closeModal={closeModal}
        />
      ) : null}
    </>
  );
};

export default App;
