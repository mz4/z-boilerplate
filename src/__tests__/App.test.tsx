import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  it("should have children", () => {
    const { debug } = render(<App />);
  });
  it("should match snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("should getByText Add Player", () => {
    render(<App />);
    expect(screen.getByText("Add Player")).toBeInTheDocument();
  });
  it("should get addPlayer getTestById", () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId("addPlayer")).toHaveTextContent("Add Player");
  });
  it("should find boxes to be in the document", () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId("boxes")).toBeInTheDocument();
  });
});
