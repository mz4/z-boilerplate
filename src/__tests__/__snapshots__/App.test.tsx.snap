// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <div
    class="Box"
    data-testid="boxes"
  >
    <div
      class="Dropdown"
    >
      <button
        class="dropbtn"
        data-testid="openMenuButton"
      >
        Menu
      </button>
      <div
        class="dropdown-content"
      />
    </div>
    <div
      class="Text"
    >
      Ciao
    </div>
  </div>
  <div
    class="Box"
  >
    <div
      class="Text"
    >
      Ciao
    </div>
  </div>
  <div>
    <h2>
      This is a title
    </h2>
    <button>
      Increase Number 
      <span>
        0
      </span>
    </button>
    <table>
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Role
          </th>
          <th>
            Age
          </th>
          <th>
            Active
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Lebron James
          </td>
          <td>
            Shooting Guard
          </td>
          <td>
            36
          </td>
          <td>
            yes
          </td>
          <td>
            <button
              data-testid="deleteplayer"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Michael Jordan
          </td>
          <td>
            Shooting Guard
          </td>
          <td>
            56
          </td>
          <td>
            no
          </td>
          <td>
            <button
              data-testid="deleteplayer"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            Gary Payton
          </td>
          <td>
            Point Guard
          </td>
          <td>
            46
          </td>
          <td>
            no
          </td>
          <td>
            <button
              data-testid="deleteplayer"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <button
    data-testid="addPlayer"
  >
    Add Player
  </button>
</div>
`;
